src/lib/markets/chain.ts

```diff
@@ -481,3 +501,58 @@ export async function fetchMarketSettlementFees({
     }>,
   )
 }
+
+export async function simulateMarketSettles({
+  lensAddress,
+  address,
+  priceCommitments,
+  marketAddresses,
+  publicClient,
+}: {
+  lensAddress: Address
+  address: Address
+  priceCommitments: UpdateDataResponse[]
+  marketAddresses: Address[]
+  publicClient: PublicClient
+}) {
+  const t = await publicClient.simulateCalls({
+    account: address,
+    calls: [
+      {
+        to: lensAddress,
+        data: encodeFunctionData({
+          abi: LensAbi,
+          functionName: 'snapshot',
+          args: [priceCommitments, marketAddresses, address],
+        }),
+        value: 0n,
+      },
+    ],
+    stateOverrides: [
+      {
+        address: lensAddress,
+        code: LensDeployedBytecode,
+        balance: maxUint256,
+      },
+    ],
+  })
+
+  const logs = t.results.map((t) => t.logs).flat()
+
+  return logs
+}
+
+export async function decodeAllEventLogs({ logs }: { logs: Awaited<ReturnType<typeof simulateMarketSettles>> }) {
+  return logs
+    .map((log) => {
+      if (!log) return
+      try {
+        const eventLog = decodeEventLog({ abi: AllEventsAbi, data: log.data, topics: log.topics })
```

Comment:
I think we should just leave the decoding up to the caller. If we want to decode here, we should return the full event, and then add a `decoded` field which contains the decoded value. If the decoding fails, the raw event still exists.

Reply:
I changed it so the util just takes 1 Log and tries to decode instead of mapping over an array